replica/table.cc

```diff
@@ -1274,6 +1274,11 @@ future<bool> table::perform_offstrategy_compaction() {
     _off_strategy_trigger.cancel();
     bool performed = false;
     co_await parallel_foreach_compaction_group([this, &performed] (compaction_group& cg) -> future<> {
+        if (!cg.maintenance_sstables()->size()) {
+            tlogger.debug("Skipping off-strategy compaction for {}.{}, No candidates were found",
+                    schema()->ks_name(), schema()->cf_name());
+            co_return;
+        }
```

Comment:
is this left over? once we agreed this logic should only exist in compaction_manager, so as not to have knowledge dupĺicated about what triggers off-strategy. we had a check like this in replica::table, but we removed it.

Reply:
I'll keep it in the compaction_manager layer in next version

Reply:
The reason it's more efficient to skip the function at the higher layers is that we actually do quite a lot, like creating the task and then acquire the semaphore just to find out we have nothing to do.

Reply:
I have an idea: add `bool needs_offstrategy_compaction()` to table_state interface, which can be used by both compaction manager and replica::table. so no knowledge duplication.

Reply:
I looked at offstrategy_compaction_task_executor::do_run() and what we can possibly do is to move the (!size) check to the beginning of the function, meaning it won't have to unnecessarily acquire the semaphore.

Reply:
yeah, let's consider this as a follow up.
I agree it's not on any hot path so maybe not worth the effort.